numpy/_core/strings.py

```diff
@@ -137,17 +137,35 @@ def multiply(a, i):
     >>> np.strings.multiply(a, i)
     array([['a', 'bb', 'ccc'],
            ['d', 'ee', 'fff']], dtype='<U3')
-           
+
     """
-    a_arr = np.asarray(a)
-    i_arr = np.asarray(i)
-    if not issubclass(i_arr.dtype.type, np.integer):
-        raise ValueError("Can only multiply by integers")
-    if a_arr.dtype.kind == "T":
-        return a_arr * i_arr
-    out_size = _get_num_chars(a_arr) * max(int(i_arr.max()), 0)
-    return _vec_string(
-        a_arr, type(a_arr.dtype)(out_size), '__mul__', (i_arr,))
+    a = np.asanyarray(a)
+    i = np.asanyarray(i)
+    a, i = np.broadcast_arrays(a, i)
+    if not np.issubdtype(i.dtype, np.integer):
+        raise TypeError(f"unsupported type {i.dtype} for operand 'i'")
+
+    # (negative number) * (any string dtype) should return an empty string
+    i = np.where(i < 0, 0, i)
+
+    # delegate to stringdtype loops that also do overflow checking
```

Comment:
Maybe have this right after the conversion of `a` to an array?

Reply:
> Maybe have this right after the conversion of `a` to an array?